crates/adapters/src/static_compile/catalog.rs

```diff
@@ -161,4 +210,324 @@ impl Catalog {
 
         self.output_batch_handles.insert(name.to_owned(), handles);
     }
+
+    /// Add an output stream that carries updates to an indexed Z-set that
+    /// behaves like a map (i.e., has exactly one key with weight 1 per value)
+    /// to the catalog.
+    ///
+    /// Creates neighborhood and quantiles circuits.  Drops the key component
```

Comment:
all these extra things should be optional. but I said that already

Reply:
I agree, and it's not hard to make it optional. We might add this feature soon.  Ideally, there would be a way to enable debugging features like this at runtime, but we don't yet have the machinery for this.